server/src/main/java/org/opensearch/index/translog/transfer/FileSnapshot.java

```diff
@@ -0,0 +1,223 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.translog.transfer;
+
+import org.opensearch.common.Nullable;
+import org.opensearch.common.io.stream.BytesStreamInput;
+import org.opensearch.common.io.stream.InputStreamStreamInput;
+import org.opensearch.core.internal.io.IOUtils;
+import org.opensearch.index.translog.BufferedChecksumStreamInput;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.channels.Channels;
+import java.nio.channels.FileChannel;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.Objects;
+
+/**
+ * Snapshot of a single file that gets transferred
+ *
+ * @opensearch.internal
+ */
+public class FileSnapshot implements Closeable {
+
+    private final String name;
+    @Nullable
+    private final FileChannel fileChannel;
+    @Nullable
+    private Path path;
+    @Nullable
+    private byte[] content;
+
+    public FileSnapshot(Path path) throws IOException {
```

Comment:
Can these constructors be private so that this cannot be extended except via the nested classes in this file?